### DISCUSSION TITLE
Is map cached?

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#4475] [Q&A]
Similar to how recently the timeline became much faster thank to caching it. Is there similar plans for the map? Now it seems a bit slow.

Beside that, once it loads it works fantastic

**Comment by danieldietzler** [comment]
Map is being reworked at the moment. For now, we don't have any performance improvements in place. However, we're considering them for the current PR #4294 or a revisit later on.

**Reply by AlliotTech** [reply]
Hmmm, I think so, and the API openstreetmap.org is not very fast as well.  
Can we try to use offline maps or support customizing other map api?

**Reply by jrasm91** [reply]
What is "slow" though? Is it the tiles loading, the api response returning, or the client-side clustering?

**Reply by AlliotTech** [reply]
Loading is slow due to network blockade in some countries

**Comment by jrasm91** [comment]
Are you talking specifically on mobile or on the web?

**Reply by sincerefly** [reply]
all device slow, in some countries. support setting openstreetmap proxy or map api should be useful

**Reply by danieldietzler** [reply]
You can already use a custom map provider by providing a custom `style.json`
